include/signalrclient/signalr_value.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace signalr
+{
+    class value
+    {
+    public:
+        enum type
+        {
+            MAP,
+            ARRAY,
+            STRING,
+            NUMBER,
+            NULL,
+            BOOL
+        };
+
+        value();
+        value(int val);
+        value(double val);
+        value(const std::string& val);
+        value(const std::vector<value>& val);
+        value(const std::string& property_name, int val);
+        value(const std::string& property_name, double val);
+        value(const std::string& property_name, const std::string& val);
+        value(const std::string& property_name, const std::vector<value>& val);
+
+        bool is_map() const;
+        bool is_number() const;
+        bool is_string() const;
+        bool is_null() const;
+        bool is_array() const;
+        bool has_property_name() const;
+
+        double as_double() const;
+        float as_float() const;
+        int as_int() const;
+        long as_long() const;
+        std::string as_string() const;
+
+        std::vector<value> get_properties() const;
```

Comment:
This seems a little odd to me. So each property is itself a `value`? I'd expect this to return something like `std::map<string, value>`...

Reply:
Maybe it should be called as_array since this is supposed to be the is_array equivalent.

Reply:
Actually we should have both